# Project data analysis

In [85]:
import numpy as np

In [83]:
import pandas as pd
import glob
import os
path='d:\moje_dokumenty\Desktop\IES\semester 11\Python\project\data\ '  # make your own path to the data
csv_path=path+"*.csv"
files=glob.glob(csv_path)      # gives you list of csv files in the directory
list_of_DFs = [pd.read_csv(file) for file in files]

In [95]:
len(list_of_DFs)

10920

In [105]:
path2='d:\moje_dokumenty\Desktop\IES\semester 11\Python\project\data2\ '  # make your own path to the data
csv_path2=path2+"*.csv"
files2=glob.glob(csv_path2)      # gives you list of csv files in the directory
list_of_DFs2 = [pd.read_csv(file) for file in files2]

In [106]:
len(list_of_DFs2)

10920

In [104]:
list_of_DFs[0]

city  year    month  Giorno  T Media    T min    T max Precip. Umidità  \
0   Bari  1985  Gennaio       1   6.6 °C   5.6 °C     8 °C    1 mm    89 %   
1   Bari  1985  Gennaio       2   5.7 °C     3 °C   9.6 °C    1 mm    91 %   
2   Bari  1985  Gennaio       3   4.2 °C     2 °C     8 °C       -    85 %   
3   Bari  1985  Gennaio       4   4.2 °C   0.8 °C     8 °C       -    77 %   
4   Bari  1985  Gennaio       5   7.7 °C   3.4 °C  15.2 °C       -    81 %   
5   Bari  1985  Gennaio       6   1.7 °C    -1 °C   4.4 °C   11 mm    94 %   
6   Bari  1985  Gennaio       7   2.4 °C  -1.2 °C     6 °C    1 mm    85 %   
7   Bari  1985  Gennaio       8   1.5 °C   0.6 °C     3 °C    3 mm    92 %   
8   Bari  1985  Gennaio       9   3.4 °C  -0.6 °C   7.4 °C       -    85 %   
9   Bari  1985  Gennaio      10   0.2 °C    -1 °C   2.2 °C       -    68 %   
10  Bari  1985  Gennaio      11   1.1 °C    -2 °C   4.4 °C       -    80 %   
11  Bari  1985  Gennaio      12   1.3 °C    -3 °C   5.6 °C       -    71 %   
12  Bari  1985  Gennaio      13   3.9 °C  -3.2 °C   9.2 °C       -    73 %   
13  Bari  1985  Gennaio      14   6.4 °C     2 °C    13 °C       -    88 %   
14  Bari  1985  Gennaio      15     9 °C     5 °C  12.2 °C     n/d    90 %   
15  Bari  1985  Gennaio      16   9.8 °C   8.4 °C  11.4 °C   12 mm    86 %   
16  Bari  1985  Gennaio      17   9.4 °C     6 °C  11.2 °C       -    76 %   
17  Bari  1985  Gennaio      18     8 °C   5.8 °C    11 °C     n/d    85 %   
18  Bari  1985  Gennaio      19   9.4 °C     7 °C  14.2 °C    4 mm    89 %   
19  Bari  1985  Gennaio      20   8.1 °C     6 °C    12 °C       -    89 %   
20  Bari  1985  Gennaio      21   9.8 °C   4.2 °C  16.2 °C       -    86 %   
21  Bari  1985  Gennaio      22  13.2 °C    10 °C  17.6 °C       -    89 %   
22  Bari  1985  Gennaio      23  13.8 °C    12 °C    17 °C       -    88 %   
23  Bari  1985  Gennaio      24  14.9 °C    10 °C    21 °C     n/d    62 %   
24  Bari  1985  Gennaio      25  10.1 °C     6 °C  13.2 °C    1 mm    79 %   
25  Bari  1985  Gennaio      26   9.4 °C     4 °C  15.8 °C       -    73 %   
26  Bari  1985  Gennaio      27   8.9 °C     6 °C    11 °C   11 mm    90 %   
27  Bari  1985  Gennaio      28   9.5 °C     8 °C    11 °C    9 mm    94 %   
28  Bari  1985  Gennaio      29   9.3 °C     9 °C  10.4 °C   12 mm    90 %   
29  Bari  1985  Gennaio      30   8.2 °C     5 °C  12.2 °C       -    87 %   
30  Bari  1985  Gennaio      31   8.8 °C     4 °C  13.2 °C       -    84 %   

    Vento Max  Raffica Fenomeni  Info  
0   20.6 km/h        -      NaN   NaN  
1   33.5 km/h        -      NaN   NaN  
2   29.4 km/h        -  Nessuno   NaN  
3   18.3 km/h        -  Nessuno   NaN  
4     37 km/h        -  Nessuno   NaN  
5   27.8 km/h        -      NaN   NaN  
6   24.1 km/h        -      NaN   NaN  
7   20.6 km/h        -      NaN   NaN  
8   20.6 km/h        -  Nessuno   NaN  
9   27.8 km/h        -  Nessuno   NaN  
10  27.8 km/h        -  Nessuno   NaN  
11  22.2 km/h        -  Nessuno   NaN  
12  14.8 km/h        -  Nessuno   NaN  
13  18.3 km/h        -  Nessuno   NaN  
14  14.8 km/h        -      NaN   NaN  
15  31.7 km/h  45 km/h      NaN   NaN  
16  27.8 km/h        -  Nessuno   NaN  
17  24.1 km/h        -      NaN   NaN  
18    13 km/h        -      NaN   NaN  
19  16.5 km/h        -  Nessuno   NaN  
20  14.8 km/h        -  Nessuno   NaN  
21  33.5 km/h  42 km/h  Nessuno   NaN  
22  35.2 km/h        -  Nessuno   NaN  
23  29.4 km/h        -      NaN   NaN  
24  16.5 km/h        -  Nessuno   NaN  
25  11.1 km/h        -  Nessuno   NaN  
26  20.6 km/h        -      NaN   NaN  
27  27.8 km/h        -      NaN   NaN  
28  29.4 km/h        -      NaN   NaN  
29  18.3 km/h        -  Nessuno   NaN  
30  24.1 km/h        -  Nessuno   NaN

In [107]:
list_of_DFs2[0]

city  year    month  Giorno  T Media     T min    T max Precip.  \
0   Bergamo  1985  Gennaio       1  -2.8 °C     -6 °C     2 °C       -   
1   Bergamo  1985  Gennaio       2  -4.9 °C     -9 °C    -1 °C       -   
2   Bergamo  1985  Gennaio       3  -2.4 °C    -15 °C     5 °C       -   
3   Bergamo  1985  Gennaio       4    -4 °C     -8 °C   1.7 °C       -   
4   Bergamo  1985  Gennaio       5  -4.4 °C     -9 °C   0.2 °C       -   
5   Bergamo  1985  Gennaio       6  -8.2 °C    -13 °C  -2.6 °C       -   
6   Bergamo  1985  Gennaio       7  -8.6 °C  -14.2 °C  -1.4 °C       -   
7   Bergamo  1985  Gennaio       8  -7.7 °C  -11.8 °C    -7 °C    1 mm   
8   Bergamo  1985  Gennaio       9  -7.4 °C    -11 °C  -3.8 °C    1 mm   
9   Bergamo  1985  Gennaio      10  -7.5 °C  -13.6 °C  -0.7 °C       -   
10  Bergamo  1985  Gennaio      11  -6.3 °C  -10.2 °C   1.2 °C       -   
11  Bergamo  1985  Gennaio      12  -5.8 °C  -11.3 °C   0.3 °C       -   
12  Bergamo  1985  Gennaio      13  -4.8 °C   -9.3 °C  -2.3 °C       -   
13  Bergamo  1985  Gennaio      14  -1.8 °C   -4.4 °C    -1 °C   15 mm   
14  Bergamo  1985  Gennaio      15  -1.9 °C     -3 °C    -1 °C    9 mm   
15  Bergamo  1985  Gennaio      16  -0.2 °C   -1.4 °C   0.7 °C       -   
16  Bergamo  1985  Gennaio      17   0.8 °C   -0.1 °C   2.4 °C   22 mm   
17  Bergamo  1985  Gennaio      18  -0.6 °C     -3 °C   0.5 °C       -   
18  Bergamo  1985  Gennaio      19   0.6 °C   -1.8 °C   2.4 °C       -   
19  Bergamo  1985  Gennaio      20  -0.8 °C   -5.3 °C     3 °C       -   
20  Bergamo  1985  Gennaio      21   1.5 °C   -0.4 °C   2.7 °C    1 mm   
21  Bergamo  1985  Gennaio      22     1 °C      0 °C     2 °C   24 mm   
22  Bergamo  1985  Gennaio      23   1.9 °C    0.8 °C   3.1 °C   10 mm   
23  Bergamo  1985  Gennaio      24   2.6 °C      1 °C   5.4 °C    3 mm   
24  Bergamo  1985  Gennaio      25   0.2 °C     -3 °C   4.2 °C       -   
25  Bergamo  1985  Gennaio      26  -0.1 °C   -2.2 °C     1 °C    5 mm   
26  Bergamo  1985  Gennaio      27  -1.7 °C     -3 °C  -0.4 °C    1 mm   
27  Bergamo  1985  Gennaio      28   1.8 °C   -3.8 °C   4.8 °C     n/d   
28  Bergamo  1985  Gennaio      29   2.3 °C     -2 °C   7.7 °C       -   
29  Bergamo  1985  Gennaio      30   0.9 °C     -4 °C   6.2 °C       -   
30  Bergamo  1985  Gennaio      31   3.1 °C   -2.4 °C   7.8 °C       -   

   Umidità  Vento Max Raffica Fenomeni  Info  
0     83 %   9.4 km/h       -  Nessuno   NaN  
1     90 %   3.5 km/h       -      NaN   NaN  
2     57 %   9.4 km/h       -  Nessuno   NaN  
3     60 %  16.5 km/h       -  Nessuno   NaN  
4     54 %   5.4 km/h       -  Nessuno   NaN  
5     60 %   7.6 km/h       -  Nessuno   NaN  
6     69 %   7.6 km/h       -  Nessuno   NaN  
7     92 %   9.4 km/h       -      NaN   NaN  
8     87 %   7.6 km/h       -  Nessuno   NaN  
9     77 %   9.4 km/h       -  Nessuno   NaN  
10    71 %  11.1 km/h       -      NaN   NaN  
11    78 %  11.1 km/h       -  Nessuno   NaN  
12    92 %          -       -      NaN   NaN  
13    93 %          -       -      NaN   NaN  
14    91 %          -       -      NaN   NaN  
15    92 %          -       -      NaN   NaN  
16    88 %   5.4 km/h       -      NaN   NaN  
17    94 %          -       -      NaN   NaN  
18    91 %          -       -  Nessuno   NaN  
19    94 %   3.5 km/h       -      NaN   NaN  
20    96 %          -       -      NaN   NaN  
21    98 %          -       -      NaN   NaN  
22    99 %          -       -      NaN   NaN  
23    78 %    13 km/h       -      NaN   NaN  
24    80 %   9.4 km/h       -  Nessuno   NaN  
25    98 %   5.4 km/h       -      NaN   NaN  
26    72 %          -       -      NaN   NaN  
27    87 %   9.4 km/h       -      NaN   NaN  
28    83 %  11.1 km/h       -      NaN   NaN  
29    85 %          -       -      NaN   NaN  
30    70 %  11.1 km/h       -  Nessuno   NaN

# Things to do: 
## analysis of the data - first of all to process them: 
*  ##  delete %, km/h etc. from the values in dataframes +  to transform strings into numerical values (floats) for appropriate variables, eg "6.6 °C" (string) into 6.6 (float)
* ## connected to the previous point, to replace missing values with monthly averages of the corresponding variable (or to drop the observation?)
* ## adding new data column to dataframes: daily range of temperature as Tmax-Tmin
* ## deleting unneccesary data columns from dataframes - Umidita, Raffica, Fenomeni,  Info (perhaps not really needed) 
* ## transliting Italian names of the variables (columns) into English, perhaps also name of the months inside of the dataframes
* ## then we can compute quarterly(?) (ie January to March, April to June...) means and variances for each city
